addons/web/static/tests/core/dropdown/dropdown.test.js

```diff
@@ -69,6 +74,24 @@ class MultiLevelDropdown extends Component {
     `;
 }
 
+class DropdownInShadowDom extends Component {
+    static components = { SimpleDropdown };
+    static props = [];
+    static template = xml`<div><SimpleDropdown/></div>`;
+}
+class ShadowDom extends Component {
+    static components = { Dropdown, DropdownItem };
+    static props = [];
+    static template = xml`<div class="shadow-root" />`;
+    setup() {
+        onMounted(() => {
+            const shadowRoot = document.querySelector(".shadow-root");
+            const shadowBody = shadowRoot.attachShadow({ mode: "open" });
+            mountWithCleanup(DropdownInShadowDom, { target: shadowBody });
+        });
+    }
+}
+
```

Comment:
You probably don't need the `DropdownInShadowDom` class, I think doing `mountWithCleanup(SimpleDropdown...` should do the trick.

Also could you move the `ShadowDom` class inside the test you wrote as it's only used there.

Reply:
https://github.com/odoo/odoo/blob/a3bb6919768c8174a926be16e87a060eac0f6028/addons/web/static/src/core/popover/popover.js#L168

The extra `Div` element is needed because of this LOC. As, if we mount the `SimpleDropdown` as it is there will be no `parentElement` of the target=Dropdown.